src/client/metadataTransfer.ts

```diff
@@ -229,23 +229,34 @@ export abstract class MetadataTransfer<
           await Lifecycle.getInstance().emitWarning('Metadata API response not parseable');
           return { completed: false };
         }
+        const retryableErrors = [
```

Comment:
should work.  I think this has gotten big/complex enough to make it it separate function (SRP) vs. being inside `poll` 

`isRetryable(error: unknown) : boolean` and then let it contain the error list and manage the type checks on what it's being passed.